internal/internal_workflow.go

```diff
@@ -480,6 +488,50 @@ func newWorkflowContext(
 	return envInterceptor, ctx, nil
 }
 
+func defaultUpdateHandler(
+	rootCtx Context,
+	name string,
+	serializedArgs *commonpb.Payloads,
+	header *commonpb.Header,
+	callbacks UpdateCallbacks,
+	goNamed func(Context, string, func(Context)),
+) {
+	env := getWorkflowEnvironment(rootCtx)
+	ctx, err := workflowContextWithHeaderPropagated(rootCtx, header, env.GetContextPropagators())
+	if err != nil {
+		callbacks.Reject(err)
+		return
+	}
+	eo := getWorkflowEnvOptions(ctx)
+	handler, ok := eo.updateHandlers[name]
+	if !ok {
+		keys := make([]string, 0, len(eo.updateHandlers))
+		for k := range eo.updateHandlers {
+			keys = append(keys, k)
+		}
+		callbacks.Reject(fmt.Errorf("unknown update %v. KnownUpdates=%v", name, keys))
+		return
+	}
+
+	args, err := decodeArgsToRawValues(handler.dataConverter, reflect.TypeOf(handler.fn), serializedArgs)
+	if err != nil {
+		callbacks.Reject(fmt.Errorf("unable to decode the input for update %q: %w", name, err))
+		return
+	}
+	input := UpdateInput{Name: name, Args: args}
+
+	goNamed(ctx, name, func(ctx Context) {
+		envInterceptor := getWorkflowEnvironmentInterceptor(ctx)
+		if err := envInterceptor.inboundInterceptor.ValidateUpdate(ctx, &input); err != nil {
+			callbacks.Reject(err)
+			return
+		}
+		callbacks.Accept()
+		success, err := envInterceptor.inboundInterceptor.ExecuteUpdate(ctx, &input)
```

Comment:
So really, if they want to, technically users _could_ transform their args in the validator and they'd see it in the update by just mutating input, right?

Inadvisable, I think, but, possible

Reply:
Yes. True in the interceptors as well. But they'd have to conform to the same type. As I understood the discussion there was also interest in changing the type(s). Simple example would be to avoid double-parsing a UUID: input to the validator is the uuid as a string but once it's parsed into a uuid.UUID (which is a [16]byte) in the validator it could stay that type as input to the update func. Would take either (1) rather a lot of variadic input/output parameter reflection hacker or (2) the validator itself returns the execute func